### ISSUE TITLE
Suppress TracerWarning by pytorch when using example_input_array

### ISSUE THREAD

**Issue by pwwang** [PyTorchLightning/pytorch-lightning#3086]
## 🚀 Feature

Suppress TracerWarning by `pytorch` when using `example_input_array`. 
These warnings won't be raised normally. However, they become very annoying when `torch.size()/shape` is to be destructed (i.e `tensor.size()[0]`) in the model.

### Motivation

It is very annoying to have the following warnings when I have code like `tensor.size()[0]` in the model and with `example_input_array` enabled:
```
TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record 
the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace 
might not generalize to other inputs!
```

### Pitch

Those warnings should be gone if they are caused by `example_input_array`

### Alternatives

It should be possible to suppress these warnings if they were raised during the detection of input/output sizes using `example_input_array`

### Additional context

See this notebook:

https://colab.research.google.com/drive/1M_YpU4z7DR_nVvAef_DF7qIC5p8aKn9O?usp=sharing

**Comment by Borda** [comment]
@awaelchli mind have look? 🌮

**Comment by awaelchli** [comment]
@pwwang can you grant me access to the colab link?

**Comment by awaelchli** [comment]
Ok, I checked and this is fixed on master by #3092
We don't log graph by default, which is very good. The tracer warning will only show if you log the graph explicitly, but we should not suppress it in my opinion.
